dhall/tests/Dhall/Test/Substitution.hs

```diff
@@ -25,7 +26,9 @@ resultDecoder :: Dhall.Decoder Result
 resultDecoder = Dhall.auto
 
 resultType :: Expr Src Void
-resultType = Dhall.expected resultDecoder
+resultType = case Dhall.expected resultDecoder of
+    Data.Either.Validation.Success x -> x
+    _ -> undefined
```

Comment:
Would it be possible have `resultType` return an `Expector` and then propagate that outward, so that we don't need the `undefined` here?

Reply:
I decided to change `resultType` to `IO (Expr Src Void)` and throw the exception there.
Is that ok?

Reply:
Yeah, that's fine.  As far as I can tell, `resultType` is only used in this test code and is not part of the public API